productions/p1.py

```diff
@@ -7,19 +7,26 @@ class P1():
     @staticmethod
     def apply(G: nx.Graph, offset = 1):
         El_node = None
-        for node in G.nodes(data=True):
-            if node[1]['label'] == 'El':
-                El_node = node
-                break
+        
+        # find isos and get pick first one
+        isomorphisms = find_isomorphisms(G, P1.left)
+        if len(isomorphisms) < 1:
+            return False
+        iso = isomorphisms[0]
```

Comment:
I find it natural that an item in the isomorphisms array would be called an isomorphism. You could call it first_isomorphism, matched_isomorphismif or propose a different name if it is confusing for you. `iso` is confusing since it has a meaning (it's not obvious that it is a shorthand).

Reply:
done

Reply:
https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.isomorphism.is_isomorphic.html#networkx.algorithms.isomorphism.is_isomorphic
![image](https://user-images.githubusercontent.com/44035398/206609473-b9c3d846-97c2-4d72-a7f8-a49165d8ab57.png)